grassroots-backend/src/GetEnvironmentVariables.ts

```diff
@@ -0,0 +1,87 @@
+import { access, readFile } from "fs/promises";
+import * as dotenv from "dotenv";
+
+// Earlier files take priority.
+function getEnvFilePaths(): string[] {
+  if (process.env.GITHUB_ACTIONS == "true") {
+    return ["../.env.test.ci", "../.env.test"];
+  }
+  if (process.env.MODE == "test") {
+    return ["../.env.test"];
+  }
+  return ["../.env.development.local", "../.env.development"];
+}
+
+/**
+ * Checks if a file exists and is accessible
+ */
+async function fileExists(filePath: string): Promise<boolean> {
+  try {
+    await access(filePath);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Reads and parses a single .env file
+ */
+async function readSingleEnvironmentFile(
+  filePath: string,
+): Promise<Record<string, string>> {
+  try {
+    const fileContent = await readFile(filePath, "utf8");
+    return dotenv.parse(fileContent);
+  } catch (error) {
+    // If file exists but can't be read/parsed, that's a real problem
+    const errorMessage = error instanceof Error ? error.message : String(error);
+    throw new Error(
+      `Failed to read environment file ${filePath}: ${errorMessage}`,
+    );
+  }
+}
+
+// Cache environment variables.
+let envVariables: Record<string, string> | undefined = {};
```

Comment:
I'm surprised that the `if (envVariables)` check doesn't complain here. I would have expected it to insist that we check whether it's equal to undefined.

Can we start envVariables as undefined, instead of {}?